internal/wal/wal_aof.go

```diff
@@ -18,43 +20,146 @@ import (
+	logDir              string
+	syncInterval        time.Duration // Interval for syncing data to disk
+	lastSyncTime        time.Time     // Last time data was synced
+	shouldFsync         bool
+	maxFileSize         int64 // Maximum size of a log segment file in bytes
+	maxSegments         int   // Maximum number of log segment files to keep
+	currentSegmentIndex int   // Index of the current log segment
+
+	rotationMode     string        // "time" or "size"
+	rotationInterval time.Duration // Time interval for rotation
+	rotationTimer    *time.Timer   // Timer for rotation
+	quit             chan struct{} // Channel to signal shutdown
+
+	isBuffered bool   // Determines if writes are buffered
+	writeMode  string // "fsync" or "default"
 }
 
-func NewAOFWAL(logDir string) (*WALAOF, error) {
-	return &WALAOF{
-		logDir: logDir,
-	}, nil
-}
+func NewAOFWAL(
+	logDir string,
+	enableFsync bool,
+	maxFileSize int64,
+	maxSegments int,
+	rotationMode string,
+	rotationInterval time.Duration,
+	isBuffered bool,
+	writeMode string,
+) (*WALAOF, error) {
+	if err := os.MkdirAll(logDir, os.ModePerm); err != nil {
+		return nil, fmt.Errorf("failed to create log directory: %w", err)
+	}
 
-func (w *WALAOF) Init(t time.Time) error {
-	slog.Debug("initializing WAL at", slog.Any("log-dir", w.logDir))
-	if err := os.MkdirAll(w.logDir, os.ModePerm); err != nil {
-		return fmt.Errorf("failed to create log directory: %w", err)
+	if rotationMode != "time" && rotationMode != "size" {
+		return nil, fmt.Errorf("invalid rotationMode: %s, must be 'time' or 'size'", rotationMode)
+	}
+
+	if rotationMode == "time" && rotationInterval <= 0 {
+		return nil, fmt.Errorf("rotationInterval must be greater than zero when rotationMode is 'time'")
 	}
 
-	timestamp := t.Format("20060102_1504")
-	path := filepath.Join(w.logDir, fmt.Sprintf("wal_%s.aof", timestamp))
+	w := &WALAOF{
+		logDir:              logDir,
+		syncInterval:        time.Millisecond * 100,
+		lastSyncTime:        time.Now(),
+		shouldFsync:         enableFsync,
+		maxFileSize:         maxFileSize,
+		maxSegments:         maxSegments,
+		currentSegmentIndex: 0,
+		rotationMode:        rotationMode,
+		rotationInterval:    rotationInterval,
+		isBuffered:          isBuffered,
+		writeMode:           writeMode,
+		quit:                make(chan struct{}),
+	}
 
-	newFile, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	files, err := filepath.Glob(filepath.Join(logDir, segmentPrefix+"*"))
 	if err != nil {
-		return fmt.Errorf("failed to open new WAL file: %v", err)
+		return nil, err
 	}
 
-	w.file = newFile
-	return nil
+	if len(files) > 0 {
+		lastSegmentID, err := findLastSegmentIndex(files)
+		if err != nil {
+			return nil, err
+		}
+		w.currentSegmentIndex = lastSegmentID
+	}
+
+	if err := w.Init(time.Now()); err != nil {
+		return nil, err
+	}
+
+	if w.rotationMode == "time" {
+		w.rotationTimer = time.NewTimer(w.rotationInterval)
+
+		go w.handleRotationTimer()
```

Comment:
Let's separate out the initiation of the WAL object and the running of the required goroutines to a `Run` method.